samples/openapi3/client/petstore/python-experimental/tests/test_discard_unknown_properties.py

```diff
@@ -36,51 +36,86 @@
 
 class DiscardUnknownPropertiesTests(unittest.TestCase):
 
-    def test_deserialize_dog_do_not_discard_unknown_properties(self):
-        """ deserialize str, Dog) with unknown properties, strict validation is enabled """
+    def test_deserialize_banana_req_do_not_discard_unknown_properties(self):
+        """
+        deserialize bananaReq with unknown properties.
+        Strict validation is enabled.
+        Simple (non-composed) schema scenario.
+        """
         config = Configuration(discard_unknown_keys=False)
         api_client = petstore_api.ApiClient(config)
         data = {
-            "class_name": "Dog",
-            "color": "black",
-            "breed": "husky",
-            "unknown_property": "a-value"
+            'lengthCm': 21.3,
+            'sweet': False,
+            # Below are additional (undeclared) properties not specified in the bananaReq schema.
+            'unknown_property': 'a-value'
+        }
+        response = MockResponse(data=json.dumps(data))
+
+        # Deserializing with strict validation raises an exception because the 'unknown_property'
+        # is undeclared.
+        with self.assertRaises(petstore_api.exceptions.ApiAttributeError) as cm:
+            deserialized = api_client.deserialize(response, ((petstore_api.BananaReq),), True)
+        self.assertTrue(re.match("BananaReq has no attribute 'unknown_property' at.*", str(cm.exception)),
+            'Exception message: {0}'.format(str(cm.exception)))
+
+
+    def test_deserialize_isosceles_triangle_do_not_discard_unknown_properties(self):
+        """
+        deserialize IsoscelesTriangle with unknown properties.
+        Strict validation is enabled.
+        Composed schema scenario.
+        """
+        config = Configuration(discard_unknown_keys=False)
```

Comment:
Are unknown_keys the same as additional properties?
How about changing this variable name to discard_additional_properties?

Reply:
No "unknown key" means the client is sending something unknown (garbage data) to the server. I.e. the property is unknown because it's not explicitly declared in the OAS document. I have added code comments to help. I think it's better handled by having code comments, one other possibility would have been to use the word "garbage" to show it's not supposed to be there, but now with the code comments it should be pretty clear.